src/lib.rs

```diff
@@ -97,59 +97,64 @@ impl From<Error> for io::Error {
     }
 }
 
-// File: 内建文件类型
 // size: 文件大小
-// offset: 文件偏移（目前指针位置），无需序列化
 // start_at: 文件在 source 中的开始位置
-// mime: 文件类型的 MIMETYPE
+#[derive(Serialize, Deserialize, Debug)]
+struct FileHeader {
+    size: u64,
+    start_at: u64,
+}
+
+impl FileHeader {
+    pub fn from_path(p: impl AsRef<Path>) -> Result<Self> {
+        Ok(Self {
+            size: std::fs::metadata(p)?.len(),
+            start_at: 0,
+        })
+    }
+}
+
+// File: 内建文件类型
+// offset: 文件偏移（目前指针位置），无需序列化
 // source: 文件源，无需序列化
 /// The file type returned by the open function implements most of the APIs of std::fs::File
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Debug)]
 pub struct File {
-    size: u64,
-    #[serde(skip)]
+    header: FileHeader,
     offset: u64,
-    start_at: u64,
-    mime: String,
-    #[serde(skip)]
-    source: Option<fs::File>,
+    source: fs::File,
 }
 
 impl File {
-    pub fn open<P: AsRef<path::Path>>(fs: &FRFS, path: P) -> Result<File> {
+    pub fn open(fs: &FRFS, path: impl AsRef<Path>) -> Result<File> {
         fs.open(path)
     }
 
     pub fn metadata(&self) -> Result<Metadata> {
-        Ok(Metadata(self.size, FileType(true)))
-    }
-
-    pub fn mimetype(&self) -> String {
-        self.mime.clone()
+        Ok(Metadata(self.header.size, FileType(true)))
     }
 }
 
 impl From<fs::File> for File {
     fn from(f: fs::File) -> Self {
         Self {
-            size: f.metadata().unwrap().len(),
+            header: FileHeader {
+                size: f.metadata().unwrap().len(),
+                start_at: 0,
+            },
             offset: 0,
-            start_at: 0,
-            mime: "".to_string(),
-            source: Some(f),
+            source: f,
         }
     }
 }
 
 impl Read for File {
     fn read(&mut self, buf: &mut [u8]) -> Result<usize> {
-        let mut file = match &self.source {
-            Some(f) => f,
-            None => return Err(Error::NotFound.into()),
-        };
-        let ptr_now = file.seek(SeekFrom::Start(self.start_at + self.offset))?;
-        let ptr_end = self.start_at + self.size;
-        let ret = file.take(ptr_end - ptr_now).read(buf)?;
+        let ptr_now = self
+            .source
+            .seek(SeekFrom::Start(self.header.start_at + self.offset))?;
+        let ptr_end = self.header.start_at + self.header.size;
```

Comment:
I think this naming is really bad. Does the header.size here refer to the size of the header or the size of the file?